code/modules/paperwork/envelope.dm

```diff
@@ -0,0 +1,81 @@
+/obj/item/weapon/paper/envelope
+	name = "envelope"
+	icon_state = "envelope_open"
+
+	var/obj/item/weapon/paper/contained_paper
+	var/open = TRUE
+	var/torn = FALSE
+	var/nano
```

Comment:
Paper folding code assumes that every result is of the `/obj/item/weapon/p_folded` type, which has a `nano` var, which paper folding code accesses.
Adding a var to envelopes with that name makes the folding code *just werk* despite them not being a subtype of `p_folded`.

I could rewrite paper folding code instead if you'd like.